thread_sync.c

```diff
@@ -1163,8 +1163,13 @@ NORETURN(static VALUE rb_queue_freeze(VALUE self));
  * call-seq:
  *   freeze
  *
- * Raises an exception:
- *   Queue.new.freeze # Raises TypeError (cannot freeze #<Thread::Queue:0x...>)
+ * The queue can't be frozen, so this method raises an exception:
+ *   Thread::Queue.new.freeze # Raises TypeError (cannot freeze #<Thread::Queue:0x...>)
+ *
+ * Unlike other collections, freezing a queue (to stop accepting #pop and #push)
```

Comment:
I don't think the documentation needs to justify the implementation, only explain it, so I would remove this second paragraph.